go/service/main.go

```diff
@@ -138,6 +138,11 @@ func (d *Service) Run() (err error) {
 		}
 	}
 
+	// Explicitly set fork type here based on KEYBASE_LABEL
+	if len(d.G().Env.GetLabel()) > 0 {
```

Comment:
I realize you suggested testing whether ForkType was set, but most commands set it to Auto already, which would seem to clash.